### ISSUE TITLE
Fix compile warnings in mantis-control-plane-client module

### ISSUE THREAD

**Issue by liuml07** [Netflix/mantis#389]
There are compile warnings in the code due to:
- deprecated APIs
- missing `@Nullable/@NotNull` annotations
- incomplete javadocs
- lambda can be replaced with method references
- raw usage of generic types
- unnecessary type parameters (like `Map<String> m = new HashMap<String>()`)
- anonymous class can be replaced with lambdas
and more.

We don't need to solve all of them, but a good coding exercise is to address most those compile warnings when possible.

This tracks the effort for the `mantis-control-plane-client` module to make the code change scoped.

**Comment by dvlpsh** [comment]
@liuml07 is this issue still up for grabs?

**Comment by liuml07** [comment]
Yes I think so. The previous PR addressed some places, but I prefer a systematic change that updates other plances in the module. Are you interested in working on this? That would be great. CC: @Andyz26
